src/user/MOM_wave_interface.F90

```diff
@@ -453,9 +466,35 @@ subroutine Update_Surface_Waves(G, GV, US, Day, dt, CS)
     if (DataSource==DATAOVR) then
       call Surface_Bands_by_data_override(day_center, G, GV, US, CS)
     elseif (DataSource==Coupler) then
-      ! Reserve for coupler hooks
+      if (.not.present(FORCES)) then 
+        call MOM_error(FATAL,"The code cannot be run with the options  "//&
+             "SURFBAND_SOURCE = COUPLER for with this driver.  If you are using a "//&
+             "wave coupled driver then check the call to update_surface_waves, otherwise"//&
+             "select another option for SURFBAND_SOURCE.")
+      endif
+      if (size(CS%WaveNum_Cen).ne.size(forces%stk_wavenumbers)) then
+        call MOM_error(FATAL, "Number of wavenumber bands in WW3 does not match that in MOM6. "//&
+             "Make sure that STK_BAND_COUPLER in MOM6 input is equal to the number of bands in "//&
+             "ww3_grid.inp, and that your mod_def.ww3 is up to date.")
+      endif
+
+      do b=1,CS%NumBands
+        CS%WaveNum_Cen(b) = forces%stk_wavenumbers(b)
+        !Interpolate from a grid to c grid
+        do II=G%iscB,G%iecB
```

Comment:
The II argument varies fastest in memory for the arrays on line 486, so for efficiency the order of the II and jj loops should be swapped, and similarly for lines 489-490.